Add PW3 Vitals

jasonacox opened the pull request:
## v0.11.0 - Add PW3 Vitals

* Add polling of Powerwall 3 Devices to pull in PW3 specific string data, capacity, voltages, frequencies, and alerts. 
* This adds mock TEPOD, PVAC and PVS compatible payloads available in vitals().

### Testing Beta

You can test the updated code by editing the `powerwall.yml` file and changing the image to the beta version (jasonacox/pypowerwall:0.11.0t64-beta3): [[UPDATED]]

```yml
    pypowerwall:
        image: jasonacox/pypowerwall:0.11.0t64-beta3
        container_name: pypowerwall
```

Restart:

```bash
./compose-dash.sh up -d
docker logs pypowerwall -f
```

Some URLs to check: 
* http://localhost:8675/vitals 
* http://localhost:8675/help (verify pw3 shows True) 
* http://localhost:8675/tedapi/components
* http://localhost:8675/tedapi/battery  

Discussion link: https://github.com/jasonacox/Powerwall-Dashboard/discussions/392#discussioncomment-10478266

jplewis2 commented:
Updated the container using your commands. No errors:
```
$ ./compose-dash.sh up -d
docker logs pypowerwall -f
Running Docker Compose...
[+] Running 10/10
 ✔ pypowerwall Pulled                                                                                                                                   10.6s 
   ✔ c6a83fedfae6 Pull complete                                                                                                                          5.5s 
   ✔ f52ce548fcd3 Pull complete                                                                                                                          5.9s 
   ✔ 594835f348c0 Pull complete                                                                                                                          6.6s 
   ✔ 062b39265da7 Pull complete                                                                                                                          6.6s 
   ✔ c4a6b685d74e Pull complete                                                                                                                          7.0s 
   ✔ a669eae28383 Pull complete                                                                                                                          7.0s 
   ✔ 453a4f4e4e68 Pull complete                                                                                                                          7.1s 
   ✔ 62a1c6eb07bc Pull complete                                                                                                                          7.7s 
   ✔ 7292013b94a1 Pull complete                                                                                                                          7.9s 
[+] Running 5/5
 ✔ Container influxdb     Running                                                                                                                        0.0s 
 ✔ Container pypowerwall  Started                                                                                                                        1.6s 
 ✔ Container weather411   Running                                                                                                                        0.0s 
 ✔ Container grafana      Running                                                                                                                        0.0s 
 ✔ Container telegraf     Started                                                                                                                        1.5s 
09/01/2024 04:08:22 PM [proxy] [INFO] pyPowerwall [0.11.0] Proxy Server [t64] - HTTP Port 8675
09/01/2024 04:08:22 PM [proxy] [INFO] pyPowerwall Proxy Started
09/01/2024 04:08:22 PM [proxy] [INFO] pyPowerwall Proxy Server - Local Mode
09/01/2024 04:08:22 PM [proxy] [INFO] Connected to Energy Gateway 192.168.91.1 (My Home)
09/01/2024 04:08:22 PM [proxy] [INFO] TEDAPI Mode Enabled for Device Vitals (full)
09/01/2024 04:08:23 PM [pypowerwall.tedapi.pypowerwall_tedapi] [WARNING] This API [get_api_sitemaster] is using mock data in tedapi mode. This message will be printed only once at the warning level.
09/01/2024 04:08:23 PM [pypowerwall.tedapi.pypowerwall_tedapi] [WARNING] This API [get_api_powerwalls] is using mock data in tedapi mode. This message will be printed only once at the warning level.
09/01/2024 04:08:23 PM [pypowerwall.tedapi.pypowerwall_tedapi] [WARNING] This API [get_api_auth_toggle_supported] is using mock data in tedapi mode. This message will be printed only once at the warning level.
```

http://localhost:8675/tedapi/battery is empty:
```{}```

Other 3 URLs look okay:
<h1>pyPowerwall [0.11.0] Proxy [t64] </h1>
<p><a href="https://github.com/jasonacox/pypowerwall/blob/main/proxy/HELP.md">Click here for API help.</a></p>
<table>
<tr><th align ="left">Stat</th><th align ="left">Value</th></tr>
<tr><td align="left">pypowerwall</td><td align ="left">0.11.0 Proxy t64</td></tr>
<tr><td align="left">mode</td><td align ="left">Local</td></tr>
<tr><td align="left">gets</td><td align ="left">571</td></tr>
<tr><td align="left">posts</td><td align ="left">0</td></tr>
<tr><td align="left">errors</td><td align ="left">0</td></tr>
<tr><td align="left">timeout</td><td align ="left">0</td></tr>
<tr><td align="left">ts</td><td align ="left">1725221555</td></tr>
<tr><td align="left">start</td><td align ="left">1725221302</td></tr>
<tr><td align="left">clear</td><td align ="left">1725221302</td></tr>
<tr><td align="left">uptime</td><td align ="left">0:04:13</td></tr>
<tr><td align="left">mem</td><td align ="left">40624</td></tr>
<tr><td align="left">site_name</td><td align ="left">My Home</td></tr>
<tr><td align="left">cloudmode</td><td align ="left">False</td></tr>
<tr><td align="left">fleetapi</td><td align ="left">False</td></tr>
<tr><td align="left">tedapi</td><td align ="left">True</td></tr>
<tr><td align="left">pw3</td><td align ="left">True</td></tr>
<tr><td align="left">tedapi_mode</td><td align ="left">full</td></tr>
<tr><td align="left">siteid</td><td align ="left">None</td></tr>
<tr><td align="left">counter</td><td align ="left">0</td></tr>
<tr><td align="left">authmode</td><td align ="left">cookie</td></tr>
<tr><td align="left">URI: /api/troubleshooting/problems</td><td align ="left">22</td></tr>
<tr><td align="left">URI: /api/sitemaster</td><td align ="left">34</td></tr>
<tr><td align="left">URI: /api/powerwalls</td><td align ="left">34</td></tr>
<tr><td align="left">URI: /api/auth/toggle/supported</td><td align ="left">14</td></tr>
<tr><td align="left">URI: /api/meters/aggregates</td><td align ="left">34</td></tr>
<tr><td align="left">URI: /api/system_status/grid_status</td><td align ="left">34</td></tr>
<tr><td align="left">URI: /api/system_status/soe</td><td align ="left">34</td></tr>
<tr><td align="left">URI: /aggregates</td><td align ="left">51</td></tr>
<tr><td align="left">URI: /soe</td><td align ="left">51</td></tr>
<tr><td align="left">URI: /strings</td><td align ="left">51</td></tr>
<tr><td align="left">URI: /alerts/pw</td><td align ="left">51</td></tr>
<tr><td align="left">URI: /pod</td><td align ="left">51</td></tr>
<tr><td align="left">URI: /freq</td><td align ="left">51</td></tr>
<tr><td align="left">URI: /temps/pw</td><td align ="left">51</td></tr>
<tr><td align="left">URI: /version</td><td align ="left">3</td></tr>
<tr><td align="left">URI: /vitals</td><td align ="left">2</td></tr>
<tr><td align="left">URI: /help</td><td align ="left">1</td></tr>
<tr><td align="left">URI: /tedapi/components</td><td align ="left">1</td></tr>
<tr><td align="left">URI: /tedapi/battery</td><td align ="left">1</td></tr>
</table>
<p>Page refresh: 2024-09-01 16:12:35.496601</p>

jasonacox commented:
> http://localhost:8675/tedapi/battery is empty:

Thanks @jplewis2 !  I fixed this and also added some missing frequency payloads for the inverter (TEPINV).  If you can, give this beta version a try:  jasonacox/pypowerwall:0.11.0t64-beta3

jasonacox commented:
https://pypi.org/project/pypowerwall/0.11.0/
